src/foundry/client/data/abstract/world-collection.d.mts

```diff
@@ -26,8 +27,10 @@ declare global {
     get directory(): Lowercase<Name> extends keyof typeof ui
       ? (typeof ui)[Lowercase<Name>]
       :
-          | (DocumentName extends foundry.CONST.FOLDER_DOCUMENT_TYPES ? DocumentDirectory<DocumentName> : never)
-          | SidebarTab
+          | (DocumentName extends foundry.CONST.FOLDER_DOCUMENT_TYPES
+              ? DocumentDirectory<Document.ImplementationClassFor<DocumentName>>
```

Comment:
I thought I recently-ish changed `DocumentDirectory` to be generic over the name rather than the class. Was I mistaken?